### PR TITLE
Add form for creating Post an add likes button

### PR DESCRIPTION
**Pull request by Yacoubou-seidou**
### Add form for creating Post an add likes button
- Create a method called `current_user` in `ApplicationController ` to make current user data available to all controllers
- Create forms to perform the following functions:
Creates a Post on behalf of the `current_user ` (use the method that you created in `ApplicationController ` ).
Create a comment on behalf of the `current_user `(use the method that you created in `ApplicationController ` ).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -6,7 +6,32 @@ def index
...
+    @post = @user.posts.build(post_params)
+    @post.author = current_user
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post was successfully created.'
+    else
+      render :new
+    end
```
Comment (danmainah) [inline app/controllers/posts_controller.rb:35]: - [ ] @Yacoubou-seidou great work adding methods to create a post and like it, however, I couldn't find a comments controller, kindly add it and ensure it has a method to create comments and both success and fail path like you did in the post controller.
- [ ] Kindly consider also adding routes to the comments controller and methods to view comments.

File: config/routes.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create] do
+     resources :likes, only: [:create]
+   end
   end
   # Defines the root path route ("/")
   # root "articles#index"
```
Comment (danmainah) [inline config/routes.rb:11]: - [ ] [optional] It would be nice to add a root to your app, it makes it easier for users to interact with the content.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Yacoubou-seidou , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Created the current_user method in your ApplicationController to make current user data available to all controllers successfully.
✔️ Added a form to add a post and a way to like a post


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by DeliceLydia — APPROVED** [review]
## Hi @Yacoubou-seidou 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Addressed changes requested



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**